### PR TITLE
fix "do_isabsolute", and add more tests.

### PR DESCRIPTION
**Pull request by tvandijck**
This should fix my problems with the DOS %env% stuff, and also does a lot of additional validation on $() tokens.

### CODE REVIEW

File: src/host/path_isabsolute.c [hunk 1]
```diff
@@ -17,13 +17,59 @@ int path_isabsolute(lua_State* L)
...
-		path[0] == '/' ||
-		path[0] == '\\' ||
-		path[0] == '$' ||
-		path[0] == '%' ||
-		(path[0] == '"' && path[1] == '$') ||
-		(path[0] == '"' && path[1] == '%') ||
-		(path[0] != '\0' && path[1] == ':')
-	);
+	char c;
+	char* closing;
```
Comment (samsinsane) [inline src/host/path_isabsolute.c]: Since this contains a pointer into the `path` string, shouldn't this also be `const`?

**Comment by samsinsane** [comment]
I reran the build due to the weird failure, it builds fine but there are a few warnings that you'll need to clean up. Here's a snippet of the warnings:
```
path_isabsolute.c: In function ‘do_isabsolute’:
path_isabsolute.c:25:2: warning: implicit declaration of function ‘isalpha’
path_isabsolute.c:34:3: warning: implicit declaration of function ‘strchr’
path_isabsolute.c:34:13: warning: incompatible implicit declaration of built-in function ‘strchr’
path_isabsolute.c:41:4: warning: implicit declaration of function ‘isdigit’
path_isabsolute.c:57:13: warning: incompatible implicit declaration of built-in function ‘strchr’
```
Other than that, :+1: LGTM

**Comment by tvandijck** [comment]
Ah, ok... will fix ;) thnx.

File: src/host/path_isabsolute.c (hunk 1 above)
Comment (tvandijck) [inline src/host/path_isabsolute.c]: yes... thnx.

**Pull request merged**